### DISCUSSION TITLE
GitHub Copilot isn't working in VS Code

### DISCUSSION THREAD

**Discussion by saxoksch** [community/community#38853] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I just installed Copilot in Ubuntu WSL on my company machine. Initially was getting this error message 
`GitHub Copilot could not connect to server. Extension activation failed: "self signed certificate in certificate chain"`
I [read](https://stackoverflow.com/questions/71367058/self-signed-certificate-in-certificate-chain-on-github-copilot) some suggestions was to install the win-ca extension for vs code and set it to append. That didn't work. I installed it both locally and in wsl but was still getting that same error. 
What did work was setting
["http.proxyStrictSSL": false in the settings.json file.](https://stackoverflow.com/questions/36506539/how-do-i-get-visual-studio-code-to-trust-our-self-signed-proxy-certificate)
Now Copilot looks like it's active. However it's still not giving suggesions.

Looking in the extension logs I found this error:
```
2022-11-11 10:53:18.322 [error] TypeError: Cannot read properties of undefined (reading 'emit')
    at ClientRequest.a (c:\Users\username\.vscode\extensions\github.copilot-1.58.7236\dist\extension.js:8:539742)
    at Object.onceWrapper (node:events:646:26)
    at ClientRequest.emit (node:events:526:28)
    at Socket.socketOnData (node:_http_client:522:11)
    at Socket.emit (node:events:526:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
```
However I can't find anybody else having this issue.

Stuff I have tried:
- Running vs code as administrator
- Uninstalling and re-installing the copilot extension
- Uninstalling and deleting the extension folder -> re-installing the copilot extension
- Restarting vs code
- Restarting the machine

Any suggestions?

**Comment by saxoksch** [comment]
I prettified the code in the js file. It looks like it's this part that is failing for some reason:
```
function a(i, a, c) {
                        var l;
                        return s.removeAllListeners(), a.removeAllListeners(), 200 !== i.statusCode ? (r("tunneling socket could not be established, statusCode=%d", i.statusCode), a.destroy(), (l = new Error("tunneling socket could not be established, statusCode=" + i.statusCode)).code = "ECONNRESET", e.request.emit("error", l), void n.removeSocket(o)) : c.length > 0 ? (r("got illegal response body from proxy"), a.destroy(), (l = new Error("got illegal response body from proxy")).code = "ECONNRESET", e.request.emit("error", l), void n.removeSocket(o)) : (r("tunneling connection has established"), n.sockets[n.sockets.indexOf(o)] = a, t(a))
                    }
``` 
Could look like the e var is being called but I can't see it's being defined. The text that comes before seems to suggest that the underlying error has something to do with the company proxy: "tunneling socket could not be established"
I'm not sure what to make of it though

**Comment by buanzo** [comment]
I just installed the WSL extension on VSCode (as started from Windows). Then, I attempted to create a new file in one of the github repos I access from within a WSL Ubuntu distro, all that went well. Copilot, which works ok on the "parent" vscode process, is not giving suggestions, as described by the OP.

Let me know if I can provide any additional information.

**Reply by kasperschnack** [reply]
Try allow-listing the following urls in the firewall and see if it works: 
default.exp-tas.com  
copilot-proxy.githubusercontent.com

**Comment by buanzo** [comment]
As a workaround, I installed Copilot Nightly on the WSL vscode window. And code suggestion works, but this is not ideal.
